drivers/sensor/adt7420/adt7420.c

```diff
@@ -0,0 +1,202 @@
+					ADT7420_CONFIG_OP_MODE(val8)) < 0) {
+			SYS_LOG_DBG("Failed to set attribute!");
+			return -EIO;
+		}
+
+		return 0;
+	case SENSOR_ATTR_UPPER_THRESH:
+		reg = ADT7420_REG_T_HIGH_MSB;
+		/* Fallthrough */
+	case SENSOR_ATTR_LOWER_THRESH:
+		if (!reg)
+			reg = ADT7420_REG_T_LOW_MSB;
+
+		value = (s64_t)val->val1 * 1000000 + val->val2;
+		value = (value / ADT7420_TEMP_SCALE) << 1;
+
+		if (adt7420_temp_reg_write(drv_data, reg, value) < 0) {
+			SYS_LOG_DBG("Failed to set attribute!");
+			return -EIO;
+		}
+
+		return 0;
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int adt7420_sample_fetch(struct device *dev, enum sensor_channel chan)
+{
+	struct adt7420_data *drv_data = dev->driver_data;
+	s16_t value;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL ||
+			chan == SENSOR_CHAN_AMBIENT_TEMP);
+
+	if (adt7420_temp_reg_read(drv_data, ADT7420_REG_TEMP_MSB, &value) < 0)
+		return -EIO;
+
+	drv_data->sample = value >> 1; /* use 15-bit only */
+
+	return 0;
+}
+
+static int adt7420_channel_get(struct device *dev,
+		enum sensor_channel chan,
+		struct sensor_value *val)
+{
+	struct adt7420_data *drv_data = dev->driver_data;
+	s32_t value;
+
+	if (chan != SENSOR_CHAN_AMBIENT_TEMP) {
+		return -ENOTSUP;
+	}
+
+	value = (s32_t)drv_data->sample * ADT7420_TEMP_SCALE;
+	val->val1 = value / 1000000;
+	val->val2 = value % 1000000;
+
+	return 0;
+}
+
+static const struct sensor_driver_api adt7420_driver_api = {
+	.attr_set = adt7420_attr_set,
+	.sample_fetch = adt7420_sample_fetch,
+	.channel_get = adt7420_channel_get,
+#ifdef CONFIG_ADT7420_TRIGGER
+	.trigger_set = adt7420_trigger_set,
+#endif
+};
+
+static int adt7420_probe(struct device *dev)
+{
+	struct adt7420_data *drv_data = dev->driver_data;
+	u8_t value;
+	int ret;
+
+	ret = i2c_reg_read_byte(drv_data->i2c, drv_data->i2c_addr,
+				ADT7420_REG_ID, &value);
+	if (ret)
+		return ret;
+
+	if (value != ADT7420_DEFAULT_ID)
+		return -ENODEV;
+
+	ret = i2c_reg_write_byte(drv_data->i2c, drv_data->i2c_addr,
+			ADT7420_REG_CONFIG, ADT7420_CONFIG_RESOLUTION |
+			ADT7420_CONFIG_OP_MODE(ADT7420_OP_MODE_CONT_CONV));
+	if (ret)
+		return ret;
+
+	ret = i2c_reg_write_byte(drv_data->i2c, drv_data->i2c_addr,
+				 ADT7420_REG_HIST, CONFIG_ADT7420_TEMP_HYST);
+	if (ret)
+		return ret;
+
+	ret = adt7420_temp_reg_write(drv_data, ADT7420_REG_T_CRIT_MSB,
+				     (CONFIG_ADT7420_TEMP_CRIT * 1000000 /
+				     ADT7420_TEMP_SCALE) << 1);
+	if (ret)
+		return ret;
+
+#ifdef CONFIG_ADT7420_TRIGGER
+	if (adt7420_init_interrupt(dev) < 0) {
+		SYS_LOG_ERR("Failed to initialize interrupt!");
+		return -EIO;
+	}
+#endif
+
+	return 0;
+}
+
+int adt7420_init(struct device *dev)
+{
+	struct adt7420_data *drv_data = dev->driver_data;
+
+	drv_data->i2c = device_get_binding(CONFIG_ADT7420_I2C_MASTER_DEV_NAME);
+	if (drv_data->i2c == NULL) {
+		SYS_LOG_DBG("Failed to get pointer to %s device!",
+			    CONFIG_ADT7420_I2C_MASTER_DEV_NAME);
+		return -EINVAL;
+	}
+
+	drv_data->i2c_addr = CONFIG_ADT7420_I2C_ADDR;
```

Comment:
Move the i2c addr to a const config structure and remove this line. You can also add the i2c_dev name to the const config structure instead of passing `CONFIG_ADT7420_I2C_MASTER_DEV_NAME` directly to `device_get_binding`

Reply:
Thanks. Makes sense to keep the config stuff in one place.